src/main/scala/EnsimePlugin.scala

```diff
@@ -400,9 +401,16 @@ object EnsimePlugin extends AutoPlugin {
       artifact = artifactFilter(classifier = Artifact.DocClassifier)
     ).toSet ++ (ensimeUnmanagedJavadocArchives in config in projectRef).gimme
 
-    // needs to take configurations into account
-    // https://github.com/ensime/ensime-sbt/issues/247
-    val deps = project.dependencies.map(_.project.project).map { n => EnsimeProjectId(n, "compile") }
+    val pattern = "([A-Za-z]+)->([A-Za-z]+)".r
+
+    val deps = project.dependencies.map { d =>
+      val name = d.project.project
+      val config = d.configuration.getOrElse("compile") match {
```

Comment:
oh, I thought it was a `Map`. If it's an option already then just remove `.get`. And yes, you're right `collect` would drop the other case. Feels wrong to duplicate the `"compile"` but we can live with it. It's convention to use uppercase pattern matches though.